genie-web/src/main/java/com/netflix/genie/web/agent/launchers/impl/LocalAgentLauncherImpl.java

```diff
@@ -51,29 +53,42 @@
     private static final String JOB_ID_OPTION = "--jobId";
     private static final String FULL_CLEANUP_OPTION = "--full-cleanup";
 
+    private static final Object MEMORY_CHECK_LOCK = new Object();
```

Comment:
Why static?

Reply:
Because though in practice this will likely be a singleton instance in our app I wanted to ensure that even if it wasn't the memory check would be atomic across all instance and that wouldn't happen if the lock was an instance variable